app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/template.rb

```diff
@@ -47,6 +49,87 @@ def raw_delete_image
     _log.error("image=[#{name}], error: #{e}")
   end
 
+  def self.encrypt_with_aes(creds)
+    cipher = OpenSSL::Cipher.new('aes-256-cbc')
+    cipher.encrypt
+
+    key  = Base64.strict_encode64(cipher.random_key)
+    iv   = Base64.strict_encode64(cipher.random_iv)
+    encr = Base64.strict_encode64(cipher.update(creds.to_json) + cipher.final)
+
+    [encr, key, iv]
+  end
+
+  def self.miq_img_by_ids(provider_id, image_id)
+    powervc = ExtManagementSystem.find(provider_id)
+    powervc.get_image_info(image_id)
+  end
+
+  def self.cos_creds(provider_id)
+    cos = ExtManagementSystem.find(provider_id)
+    cos.cos_creds
+  end
+
+  def self.node_creds(provider_id)
+    powervc = ExtManagementSystem.find(provider_id)
+    endp = powervc.node_endpoint
+    auth = powervc.node_auth
+
+    return endp.hostname, auth.userid, auth.password
+  end
+
+  def self.image_ems_ref(bucket_id)
+    MiqTemplate.find(bucket_id).uid_ems
+  end
+
+  def self.bucket_name(bucket_id)
+    CloudObjectStoreContainer.find(bucket_id).name
+  end
+
+  def self.raw_import_image(ext_management_system, options = {})
+    session_id = DateTime.now.strftime('%Q')
+
+    location, user, password = node_creds(options['src_provider_id'])
+    hosts = ["[powervc]\n#{location}\n[powervc:vars]\nansible_connection=ssh\nansible_user=#{user}\nansible_ssh_pass=#{password}"]
+
+    cos_credentials = cos_creds(options['obj_storage_id'])
```

Comment:
Think this would be more readable if you did something like `instance_id, api_key = cos_creds()` instead of `cos_credentials[0], cos_credentials[1]`

Reply:
Sure, I will change.

Reply:
done